stl/inc/span

```diff
@@ -432,7 +437,7 @@ public:
 #if _CONTAINER_DEBUG_LEVEL > 0
         _STL_VERIFY(_Mysize <= dynamic_extent / sizeof(element_type),

```

Comment:
Is this check redundant with similar checks in the constructor?  Is it even possible to construct a span that would violate this?

Reply:
It is difficult, but possible, to violate this precondition. We have a "death test" for this check here:

https://github.com/microsoft/STL/blob/fc15609a0f2ae2a134c34e7c9a13977994f37367/tests/std/tests/P0122R7_span_death/test.cpp#L263-L266

We should probably move this check into every constructor. I've already written the hardening changes and this will be tangled up with them (the overflow precondition shouldn't be part of hardening, but it should interact with the `if constexpr` for static vs. dynamic extents), so instead of altering this prelude PR, I've recorded a todo and will perform this enhancement in the hardening PR as a special exception. Thanks!

Reply:
I've implemented and tested this change - it's a nice improvement.

It wasn't as tangled up with hardening as I feared - I found that it was better to perform in `span`'s base class which has different cases for static vs. dynamic extents. However, it's still slightly tangled up with CDL removal and test changes. Although I could backport it to the prelude PR if necessary, rebasing the changes would be annoying enough that I still plan to leave it in the next PR.

Thanks again!